### ISSUE TITLE
c.Ctx.Input.RequestBody empty value

### ISSUE THREAD

**Issue by goldalworming** [astaxie/beego#1831]
this is happen after I update beego to 1.6.1. 

this is my code
	func (c *AuthController) Login() {

		var data models.User
		if err := json.Unmarshal(c.Ctx.Input.RequestBody, &data); err != nil {
			c.Ctx.Output.SetStatus(400)
			fmt.Println(c.Ctx.Input.RequestBody)
			c.Ctx.Output.Body([]byte("empty title"))
			return
		}
		res := struct{ 	
			Data []*models.User 
		}{}

		var cond *orm.Condition
		cond = orm.NewCondition()

		cond = cond.And("username", data.Username)
		cond = cond.And("password", data.Password)
		lim :=1

		var qs orm.QuerySeter
		qs = orm.NewOrm().QueryTable("user").SetCond(cond).Limit(lim)
		_,err := qs.All(&res.Data)
		if err != nil {
			c.Ctx.Output.SetStatus(400)
			c.Ctx.Output.Body([]byte(err.Error()))
			return
		}
		result := struct{ 	
			Success bool
			Message string
		}{}	
	}

curl -X POST  http://lo0/login -d '{"username" : "admin", "password":"admin"}'
empty title

lmvm@lm17 ~/go/src/github.com/goldalworming/auth $ go run main.go 
[] <----fmt.Println return this empty array

**Comment by astaxie** [comment]
have you turn on the `copyrequestbody = true`? it looks the `c.Ctx.Input.RequestBody` is empty

**Comment by goldalworming** [comment]
appname = github.com/goldalworming/auth
httpport = 8080
runmode = prod
CopyRequestBody = true
AutoRender = false
TemplateLeft = [[
TemplateRight = ]]

**Comment by astaxie** [comment]
@goldalworming could you print `BConfig.CopyRequestBody` inside the Login()

**Comment by goldalworming** [comment]
if err := json.Unmarshal(c.Ctx.Input.RequestBody, &data); err != nil {
		c.Ctx.Output.SetStatus(400)
		fmt.Println(c.Ctx.Input.RequestBody)
		fmt.Println(beego.BConfig.CopyRequestBody)
		c.Ctx.Output.Body([]byte("empty title"))
		return
	}

lmvm@lm17 ~/go/src/github.com/goldalworming/auth $ go run main.go 
[]
false

why it return false?

**Comment by astaxie** [comment]
I don't know what happen with you config.

**Comment by goldalworming** [comment]
ok..I put 
  beego.BConfig.CopyRequestBody = true
in main.go

**Comment by goldalworming** [comment]
oh no..all config value not readed by go

I change the port to 8081 but it still work in 8080..what should I do?

**Comment by goldalworming** [comment]
it happen in 1.6.1..all my apps cannot read conf
do you now how to downgrade to 1.6.0 ?

**Comment by astaxie** [comment]
@goldalworming I don't know why you have this problem. you can try `bee api xxxx` in your local.

But anyway if you want to degrade to 1.6.0. Here it is
```
$ cd $GOPATH/src/github.com/astaxie/beego
$ git checkout 1.6.0
```

**Comment by goldalworming** [comment]
lmvm@lm17 ~/go/src/github.com/astaxie/beego $ git checkout 1.6.0
error: pathspec '1.6.0' did not match any file(s) known to git.

**Comment by astaxie** [comment]
git checkout v1.6.0

**Comment by astaxie** [comment]
@goldalworming please check https://github.com/astaxie/beego/issues/1830#issuecomment-200705596. it should be works in 1.6.1

**Comment by goldalworming** [comment]
this line disappear in 1.6.1
go run main.go 
2016/03/24 14:44:05 [asm_amd64.s:1721][I] http server Running on :8082

**Comment by goldalworming** [comment]
CopyRequestBody never go anywhere...beego 1.6.1 cannot read all my app.conf value include port
![screenshot from 2016-03-24 14 46 51](https://cloud.githubusercontent.com/assets/440256/14011227/82241456-f1cf-11e5-8377-9f8087307b95.png)

**Comment by goldalworming** [comment]
I think it still a bug in 1.6.1 that beego cannot read app.conf

**Comment by astaxie** [comment]
@goldalworming bee application not support `go run`. you need to use `bee run`

**Comment by goldalworming** [comment]
I think the problem is in 1.6.1. it  usually run well in 1.6.0 using go run or go build.
and this line not shown in 1.6.1 .
2016/03/24 14:44:05 [asm_amd64.s:1721][I] http server Running on :8082

**Comment by astaxie** [comment]
@youngsterxyf could you help to check @goldalworming problem? Whether it works in beego1.6.0 with go run. while not working in beego1.6.1

**Comment by goldalworming** [comment]
now..everything work fine 
the code same as above 
	var data models.User
	if err := json.Unmarshal(c.Ctx.Input.RequestBody, &data); err != nil {
		c.Ctx.Output.SetStatus(400)
		c.Ctx.Output.Body([]byte("empty title"))
		return
	}
	fmt.Println(c.Ctx.Input.RequestBody)
	fmt.Println(beego.BConfig.CopyRequestBody)
	res := struct{ 	
		// Status string
		Data []*models.User 
	}{}
lmvm@lm17 ~/go/src/github.com/goldalworming/auth $ go run main.go 
2016/03/24 15:44:49 [asm_amd64.s:1721][I] http server Running on :8080
[123 34 117 115 101 114 110 97 109 101 34 32 58 32 34 97 100 109 105 110 34 44 32 34 112 97 115 115 119 111 114 100 34 58 34 97 100 109 105 110 34 125]
true

curl -X POST  http://localhost:8080/login -d '{"username" : "admin", "password":"admin"}'
{"Success":false,"Message":"Combination not found"}

**Comment by goldalworming** [comment]
$ bee version
bee   :1.4.1
beego :1.6.0
Go    :go version go1.5.3 linux/amd64

**Comment by youngsterxyf** [comment]
@astaxie @goldalworming Beego 1.6.1 does not support `go run`，because when running beego app，the current dir will be changed to the dir where the app executable file lies in, when using `go run`, this dir is a temporary dir.

**Comment by goldalworming** [comment]
is it the problem why beego cannot read conf/app.conf folder?
what's happen when I build it? and how to build the code?

**Comment by cch123** [comment]
@goldalworming just run
```
go build
```
in your project root dir

**Comment by yate1996** [comment]
go build

**Comment by astaxie** [comment]
@goldalworming yea, it looks we make a wrong improvement in 1.6.1. We still in discussion about this problem. Hopefully we will fix it in 1.6.2.

**Comment by astaxie** [comment]
fixed in beego1.7.0
